delta-lake/src/main/java/org/apache/iceberg/delta/DeltaLakeToIcebergMigrationActionsProvider.java

```diff
@@ -18,17 +18,32 @@
  */
 package org.apache.iceberg.delta;
 
+import org.apache.iceberg.catalog.TableIdentifier;
+
 /**
  * An API that should be implemented by query engine integrations that want to support snapshotting
  * from Delta Lake table to Iceberg table.
  */
 public interface DeltaLakeToIcebergMigrationActionsProvider {
 
-  /** Initiates an action to snapshot an existing Delta Lake table to an Iceberg table. */
-  default SnapshotDeltaLakeTable snapshotDeltaLakeTable(String sourceTableLocation) {
-    return new BaseSnapshotDeltaLakeTableAction(sourceTableLocation);
+  /**
+   * Initiates an action to snapshot an existing Delta Lake table to an Iceberg table.
+   *
+   * @param tableIdent the identifier of the Iceberg table
+   * @param sourceTableLocation the location of the Delta Lake table
+   * @return a {@link SnapshotDeltaLakeTable} action
+   */
+  default SnapshotDeltaLakeTable snapshotDeltaLakeTable(
+      String tableIdent, String sourceTableLocation) {
+    return new BaseSnapshotDeltaLakeTableAction(sourceTableLocation)
```

Comment:
I think it may be helpful to include `tableIdent` as an argument. First of all, the table identifier is required for the action to be executed. Secondly, under Spark Context, it makes it easier to infer both the catalog instance and the table identifier from the `tableIdent` string (ex. `<catalog name>.db.table`). Also, it would make it easier to align with the `ActionProvider` interfaces which `SparkActions` implement:

https://github.com/apache/iceberg/blob/ce6c95025d33aac663ecdfb99ff92f2d71731263/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SparkActions.java#L51-L58